pkg/metrics/queue.go

```diff
@@ -131,3 +131,11 @@ func (m *QueueMetrics) SetQueuePendingResourceMetrics(resourceName string, value
 func (m *QueueMetrics) AddQueuePendingResourceMetrics(resourceName string, value float64) {
 	m.ResourceMetrics.With(prometheus.Labels{"state": "pending", "resource": resourceName}).Add(value)
 }
+
+func (m *QueueMetrics) SetQueuePreemptingResourceMetrics(resourceName string, value float64) {
+	m.ResourceMetrics.With(prometheus.Labels{"state": "preempting", "resource": resourceName}).Set(value)
+}
+
+func (m *QueueMetrics) AddQueuePreemptingResourceMetrics(resourceName string, value float64) {
```

Comment:
This method doesn't seem to be called from anywhere

Reply:
Correct, it was added for parity with the other metrics methods. It seemed appropriate to match the existing patterns.